zokrates_core/src/proof_system/bn128/utils/ffi.rs

```diff
@@ -20,14 +20,27 @@ extern "C" {
 }
 
 impl Buffer {
-    pub fn from_vec(v: &mut Vec<u8>) -> Buffer {
-        let length = v.len() as i32;
+    pub unsafe fn from_vec(v: &Vec<u8>) -> Buffer {
```

Comment:
can we make this into an unsafe block of minimal size?

Reply:
This method does not need to be unsafe, it does use unsafe methods under the hood though. Removed the unsafe keyword for clarity.